lms/djangoapps/certificates/tests/test_api.py

```diff
@@ -31,8 +31,53 @@
 FEATURES_WITH_CERTS_ENABLED['CERTIFICATES_HTML_VIEW'] = True
 
 
+class WebCertificateTestMixin(object):
+    """
+    Mixin with helpers for testing Web Certificates.
+    """
+    @contextmanager
+    def _mock_passing_grade(self):
+        """Mock the grading function to always return a passing grade. """
+        symbol = 'courseware.grades.grade'
+        with patch(symbol) as mock_grade:
+            mock_grade.return_value = {'grade': 'Pass', 'percent': 0.75}
+            yield
```

Comment:
I don't think I've seen wrapped context managers like this.  I like the pattern.